libs/auth/scripts/configure_java_card.ts

```diff
@@ -0,0 +1,323 @@
+          PUT_DATA_ADMIN.KEY_ATTRIBUTE_TAG,
+          Buffer.from([PUT_DATA_ADMIN.KEY_ATTRIBUTE_NONE])
+        ),
+      ])
+    ),
+  });
+}
+
+function configureDataObjectSlotCommandApdu(objectId: Buffer): CommandApdu {
+  return new CommandApdu({
+    cla: { secure: true },
+    ins: PUT_DATA_ADMIN.INS,
+    p1: PUT_DATA_ADMIN.P1,
+    p2: PUT_DATA_ADMIN.P2,
+    data: constructTlv(
+      PUT_DATA_ADMIN.CREATE_OBJECT_TAG,
+      Buffer.concat([
+        constructTlv(PUT_DATA_ADMIN.ID_TAG, objectId),
+        constructTlv(
+          PUT_DATA_ADMIN.ACCESS_MODE_OVER_CONTACT_INTERFACE_TAG,
+          Buffer.from([PUT_DATA_ADMIN.ACCESS_MODE_ALWAYS])
+        ),
+        constructTlv(
+          PUT_DATA_ADMIN.ACCESS_MODE_OVER_CONTACTLESS_INTERFACE_TAG,
+          Buffer.from([PUT_DATA_ADMIN.ACCESS_MODE_NEVER])
+        ),
+      ])
+    ),
+  });
+}
+
+function runAppletConfigurationCommands() {
+  sectionLog('🔧', 'Running applet configuration commands...');
+
+  const apdus = [
+    // Set PIN
+    new CommandApdu({
+      cla: { secure: true },
+      ins: CHANGE_REFERENCE_DATA_ADMIN.INS,
+      p1: CHANGE_REFERENCE_DATA_ADMIN.P1,
+      p2: CHANGE_REFERENCE_DATA_ADMIN.P2_PIN,
+      data: construct8BytePinBuffer(DEFAULT_PIN),
+    }),
+
+    // Set PUK
+    new CommandApdu({
+      cla: { secure: true },
+      ins: CHANGE_REFERENCE_DATA_ADMIN.INS,
+      p1: CHANGE_REFERENCE_DATA_ADMIN.P1,
+      p2: CHANGE_REFERENCE_DATA_ADMIN.P2_PUK,
+      data: PUK,
+    }),
+
+    // Configure max invalid PIN attempts
+    new CommandApdu({
+      cla: { secure: true },
+      ins: PUT_DATA_ADMIN.INS,
+      p1: PUT_DATA_ADMIN.P1,
+      p2: PUT_DATA_ADMIN.P2,
+      data: constructTlv(
+        PUT_DATA_ADMIN.UPDATE_CONFIG_TAG,
+        constructTlv(
+          PUT_DATA_ADMIN.PIN_POLICY_TAG,
+          constructTlv(
+            PUT_DATA_ADMIN.MAX_INVALID_PIN_ATTEMPTS_OVER_CONTACT_INTERFACE_TAG,
+            Buffer.from([MAX_INVALID_PIN_ATTEMPTS])
+          )
+        )
+      ),
+    }),
+
+    // Configure key slots
+    configureKeySlotCommandApdu(
+      CARD_VX_CERT.PRIVATE_KEY_ID,
+      // This doesn't mean that the private key itself can be accessed, just that it can always be
+      // used for signing operations, without a PIN
+      PUT_DATA_ADMIN.ACCESS_MODE_ALWAYS
+    ),
+    configureKeySlotCommandApdu(
+      CARD_VX_ADMIN_CERT.PRIVATE_KEY_ID,
+      PUT_DATA_ADMIN.ACCESS_MODE_PIN_GATED
+    ),
+
+    // Configure data object slots
+    configureDataObjectSlotCommandApdu(CARD_VX_CERT.OBJECT_ID),
+    configureDataObjectSlotCommandApdu(CARD_VX_ADMIN_CERT.OBJECT_ID),
+    configureDataObjectSlotCommandApdu(VX_ADMIN_CERT_AUTHORITY_CERT.OBJECT_ID),
+    ...GENERIC_STORAGE_SPACE.OBJECT_IDS.map((objectId) =>
+      configureDataObjectSlotCommandApdu(objectId)
+    ),
+  ];
+
+  const apduStrings = apdus.map((apdu) => apdu.asHexString(':'));
+  console.log(
+    'Sending the following APDUs over a GlobalPlatform Secure Channel:'
+  );
```

Comment:
Are there any secrets in this APDUs? Wondering if it's safe to log them (assuming log output might not be kept private). Nothing pokes out as obviously secret but figured I'd ask to make sure

Reply:
Thanks for sanity checking! Nothing secret in the APDUs. In fact, once we update to use the TPM-held private key in prod, there'll be nothing sensitive in this script. All the sensitive private keys will never leave their chips!